pom.xml

```diff
@@ -93,18 +93,19 @@
       <optional>true</optional>
     </dependency>
     <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-context-support</artifactId>
+      <version>4.3.11.RELEASE</version>
+      <optional>true</optional>
+      <scope>provided</scope>
```

Comment:
What does optional do when used with provided?
I've only used optional with compile scope artifacts.

Reply:
Thanks for your comment.

I think the Spring's artifact not required for using the core feature of Passy, therefore I've set true for the optional flag. And I've set the provided at the scope to indicates that user can select Spring version.
What do you think?

Reply:
But It might be better only set `optional=true`.
[Maven document](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope) says about `provided` as follows:

> This is much like compile, but indicates you expect the JDK or a container to provide the dependency at runtime. For example, when building a web application for the Java Enterprise Edition, you would set the dependency on the Servlet API and related Java EE APIs to scope provided because the web container provides those classes. This scope is only available on the compilation and test classpath, and is not transitive.

Reply:
@dfish3r, Should I remove the scope element?

Reply:
No, let's stick with `provided` for now. I see other projects marking dependencies this way, so we'll go with it.